C4_Deployment.puml

```diff
@@ -2,22 +2,71 @@
 ' uncomment the following line and comment the first to use locally
 ' !include C4_Container.puml
 
+' Colors
+' ##################################
+
+!global $NODE_FONT_COLOR = "#000000"
+!global $NODE_BORDER_COLOR = "#A2A2A2"
+
 ' Styling
 ' ##################################
 
 skinparam rectangle<<node>> {
     Shadowing false
     StereotypeFontSize 0
-    FontColor #000000
-    BorderColor #A2A2A2
+    FontColor $NODE_FONT_COLOR
+    BorderColor $NODE_BORDER_COLOR
     BorderStyle solid
 }
 
+' Special
+' ##################################
+
+' PlantUML supports no automatic line breaks of "PlantUML containers" (C4 Deployment_Node is a "PlantUML container")
+' therefore Deployment_Node() implements an automatic line break based on spaces (like in all other objects).
+' If a $type contains \n then the these are used (and no automatic space based line breaks are done)
+' $NODE_TYPE_MAX_CHAR_WIDTH defines the automatic line break position
+!global $NODE_TYPE_MAX_CHAR_WIDTH = 35
+
 ' Elements
 ' ##################################
 !unquoted function Deployment_Node($alias, $label)
 !return 'rectangle "=='+$label+'" <<node>> as '+$alias
 !endfunction
+
 !unquoted function Deployment_Node($alias, $label, $type)
-!return 'rectangle "=='+$label+'\n<size:'+$TECHN_FONT_SIZE+'>['+$type+']</size>" <<node>> as '+$alias
+!$multiLine = ""
+!if (%strpos($type, "\n") >= 0)
+  !while (%strpos($type, "\n") >= 0)
+    !$brPos = %strpos($type, "\n")
+    !$multiLine = $multiLine + %substr($type, 0, $brPos) + '</size>\n<size:'+$TECHN_FONT_SIZE+'>'
+    !$type = %substr($type, $brPos+2)
+  !endwhile
+!else
```

Comment:
I find the addition of `\n` useful in the `$type` parameter but I don't think there needs to be a default `$NODE_TYPE_MAX_CHAR_WIDTH` mostly because you can have multiple containers in one deployment node and the type parameter has plenty of space on one line. (Yes, an argument can be made against `skinparam wrapWidth 200` as well, I can't say I like that either)

Letting the user decide he wants a new line is enough for me. Let's see what others think.

Reply:
You can "deactivate" the automatic line break with something like "!$NODE_TYPE_MAX_CHAR_WIDTH=2000"
e.g.
[2000 char width](http://www.plantuml.com/plantuml/png/rLPjJ-D64FwUNp4NLU95G212rQbIfGj9Ua09YN0eTr8bQxmxIRQiTwtTTO1MzzyxQmU28QSUTv-Q3vPtFC-yFFEo-U03kb3cklLE6Q5BIR08eV2_T3eExmxcAYpAhFJaX3M1J3WGDk_SAYSMFYmw_UFzYKOJhaSN7Rhdxv8a2zC15Te-v56_YBQte3GCZ0882uAPrThUAJC7hGm16WcDpyfvVhF0Za5RWLe_i8sd23TuQBtxwN8y6ARJpvDXEkfzIlkdlQlqvcmmFVsrUtXus6fjNVG-Zw-dQIz9az-7q_xfzawh5h7Qz3Iz67uSNWvOs0egQ8Ar1nWed3lCEMm7Pqo3CnJW1Crjp21vy85os8SBjQHMQmrBBwsaxKABFMYVg7a503wsNvn1Oa0G50stT_vf0VzUMf66GPcQPupPnqBjxkx-6UxVHrFNMMb22KS_7nmUmyKqUJInoGmMA1R4e5xr0bER2mpjbm9uVt2_DbF_-bn_L8OS6r4HC3c3Nb5enVhAccZZ79TOrJ8fN6Hcz4S_YYVEBfKa_vgw6NT51ADMuG6MYj5mdemlEwVJwIG1Td7GtdcAuakhUcxaPqMDp0wFdX4LZs-bo3eKEb8qhbwuhXEdSdGFxRNig2k-IiyWsvOPgmumO8QU8koADDPvmgCO4gOxn_Wr2TOnEt78mD5S-U0gRL266owltlTWWNv14h0C2uvrnJe8Hp8UKVizG278Uz1spWSAuktaTPlaTR-FlEvpzX8IriWtyjVzdn6OFIVkZh9wF8-_RpoxFpYVx9y1DvHzutmEIAibEL-jNXzONS1gsqVjA7uzkfuVcg30EG6ktMnYxClhfPZR66A_zC7cvBOyi42nEKbB9QgAnz3PxJW1ww1dfBDAnfIRgTOmLX_LU4P5y2VCaJDiB1y4hN8LI88lCqy1x2nczbzBALWGgp1zt50HermpNavwKvB2vaKpooWfuvQF4IidhBUp03VAI7ldOq1rrZcASVArco1ph4jkLU6J-XGDVhGsLgI_S3PdIaRsRwKrmc_AqSpUxwri9pXZQYAf8oLS7S1GpkdHNPDVNs2ShxhM4wxrffPAX5D5tIezCoyrkYRnMlyOrzpFiCgQcLutrHNffmQexwGHthB3U3C9zkhPO8paBszDVQCmEe8ugxSXqkjwg1ZOc1TRPINM1NuMwifNvKPcrGv3wM76rQcekdCgqAF3yy59lmdeVZCYlUA5kORKvI5V_wi8bfK-a95bhly5)

Reply:
yes, noticed that, but removing it just seemed like a simpler solution. Nevertheless I would like someone else to express an opinion so I don't decide stuff like this on my own. If it's up to me, I always prefer the simplest thing.

Reply:
I agree the source is simpler. 
1. But if I write a puml-generator  then model data should be enough and "no" layout specific data should be added anymore [like my old version, line 623](https://github.com/structurizr/dotnet-extensions/blob/master/Structurizr.PlantUML/IO/C4PlantUML/C4PlantUmlWriter.cs).
2. A too long line disturbs the layout more than 2 lines (if it is really unwanted)